core/src/main/java/org/infinispan/protostream/TagHandler.java

```diff
@@ -13,8 +15,10 @@
 
    /**
     * Start of top level message. Do whatever required init here.
+    *
+    * @param descriptor The {@link GenericDescriptor} for the top level Enum or Message. For primitives it is null.
     */
-   default void onStart() {
+   default void onStart(AnnotatedDescriptor descriptor) {
```

Comment:
Same for jdg. All jdg versions previous to the upcoming 7.2 will be stuck with a pre 4.2 version of protostream. Only jdg 7.2 will start using 4.2 and will need to adapt to this change.